packages/react/src/components/Provider/Provider.tsx

```diff
@@ -33,6 +33,7 @@ export interface ProviderProps extends ChildrenComponentProps {
   renderer?: Renderer
   rtl?: boolean
   disableAnimations?: boolean
+  enableCaching?: boolean
```

Comment:
1. Let's **enable** it by default as it's our strategy to have caching
2. Let's use bitwise operations for flags (https://blog.rinatussenov.com/juggling-bits-in-javascript-bitmasks-128ad5f31bed) as we obviously want to add more settings like caching variables and disabling Fela plugins

Reply:
1. Will enable it by default.
2. We can't do it like that, because we are merging with above theme, so if something is not defined is not the same as if something is false. For undefined we are taking the value from the parent, for false we are overriding. Maybe we should make option performance where we can add all these different options in the future, like
`<Provider performance={{useStylesCache: true, useVariablesCache: true}}></Provider>`
Should we change it like this, or leave it as a first class prop on the Provider?

Reply:
```
performance={{useStylesCache: true, useVariablesCache: true}}>
```

Is okay, let's do it 👍

Reply:
I've renamed tham to `enableVariablesCache` and `enableStylesCache`, as we already have disable/emable naming in the context